File: 852. Peak Index in a Mountain Array/Solution.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int left = 0, right = arr.length - 1;
        while (left < right) {
            int mid = (left + right) / 2;
            if (arr[mid] > arr[mid + 1] && arr[mid] > arr[mid - 1]) {
                left = mid;
                break;
            }
            if (arr[mid] > arr[mid + 1]) {
                right = mid;
            } else {
                left = mid;
            }
        }
        return left;
    }
}

File: 852. Peak Index in a Mountain Array/Solution.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        left, right = 0, len(arr) - 1
        while left < right:
            mid = (left + right) // 2
            if arr[mid] > arr[mid + 1] and arr[mid] > arr[mid - 1]:
                left = mid
                break
            if arr[mid] > arr[mid + 1]:
                right = mid
            else:
                left = mid
        return mid
